docs/grammar.txt

```diff
@@ -100,6 +105,8 @@ Identifier = [a-zA-Z_] [a-zA-Z_0-9]*
 HexNumber = '0x' [0-9a-fA-F]+
 DecimalNumber = [0-9]+
 
+ElementaryTypeNameExpression = ElementaryTypeName '(' expression ')'
```

Comment:
It is kind of documented here: https://solidity.readthedocs.io/en/develop/miscellaneous.html#esoteric-features
We are currently rewriting that section and it will probably turn to something similar to your example.